src/OpenTelemetry.Contrib.Instrumentation.Azure/Implementation/AzureSdkDiagnosticListener.cs

```diff
@@ -28,14 +27,15 @@ internal class AzureSdkDiagnosticListener : ListenerHandler
         internal const string ActivitySourceName = "AzureSDK";
         internal const string ActivityName = ActivitySourceName + ".HttpRequestOut";
         private static readonly Version Version = typeof(AzureSdkDiagnosticListener).Assembly.GetName().Version;
-        private static readonly ActivitySource AzureSDKActivitySource = new ActivitySource(ActivitySourceName, Version.ToString());
+        private readonly ActivitySourceAdapter activitySource;
 
         // all fetchers must not be reused between DiagnosticSources.
         private readonly PropertyFetcher<IEnumerable<Activity>> linksPropertyFetcher = new PropertyFetcher<IEnumerable<Activity>>("Links");
 
-        public AzureSdkDiagnosticListener(string sourceName)
+        public AzureSdkDiagnosticListener(string sourceName, ActivitySourceAdapter activitySource)
```

Comment:
these instrumentations were left in main repo *without* any tests, and we just migrated it as is from main repo. Without unit tests, we cannot validate if this is doing the right thing.